src/me/mrCookieSlime/Slimefun/SlimefunGuide.java

```diff
@@ -80,6 +83,10 @@ public static ItemStack getItem() {
 		return getItem(BookDesign.CHEST);
 	}
 
+	public void inventoryClickEvent(InventoryClickEvent e) {
```

Comment:
Please follow our naming conventions.

'onInventoryClick' for example.

Reply:
I can't change the function name.  At that point this fix ceases to work.  It works because it's using the bukkit name for the function, I believe.  While I understand you have your own naming convention, trying to rewrite how bukkit deals with inventory results in exactly these sorts of glitches.  The aforementioned duplication issue exists because individual items were left out of whatever loop that assigned override actions for clicks on them... but even if those were put back in, the underlying problem that "the book chest display is being handled as a bukkit inventory" still exists so... blank slots in the book's "inventory" can still have items put into them... making the book gui an unintentional trash can, since items can't always be taken back once placed into those empty slots.    I would take a further step of removing all events that aren't hyperlink actions from all the menu items from the code to see if that further speeds up the loading of the guide, but I thought I'd just do a small thing for now.  Well, use this commit or don't, but I don't know how else to make it work.   But changing the name for me broke the fix so... I'm thinking that can't be done in this case.  Let me know if I'm wrong.

Reply:
The method name plays no role, you are extending. Not overwriting.
Also, the event should not work because you are missing an EventHandler annotation and a registration of this Listener.
Speaking of which, this being a static utility class, should not contain any Event.
Also cancelling it based on no statements, would result in no Inventory on the Server working at all.

Did you really test this?
Because normally your code should not even work, and if it worked, it would break everything.